src/Sarif.Viewer.VisualStudio/Fixes/FixSuggestedActionsSource.cs

```diff
@@ -97,7 +97,7 @@ public async Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet req
         {
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-            return GetSuggestedActions(requestedActionCategories, range, cancellationToken).Any();
+            return GetSuggestedActions(requestedActionCategories, range, cancellationToken)?.Any() == true;
```

Comment:
>GetSuggestedActions [](start = 19, length = 19)

The alternative is to have GetSuggestedActions return Enumerable.Empty<>> when there are no results removing the need for the "== true" comparison.

It's typically not great practice to have something return null when it's return is IEnumerable

Reply:
Sure, I'll do that in the PR I'm about to re-raise ("starting point for background analysis").

---
In reply to: [520966016](https://github.com/microsoft/sarif-visualstudio-extension/pull/265#discussion_r520966016) [](ancestors = 520966016)